src/mumble/widgets/EventFilters.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_WIDGETS_EVENTFILTERS_H_
+#define MUMBLE_MUMBLE_WIDGETS_EVENTFILTERS_H_
+
+#include <initializer_list>
+#include <set>
+
+#include <QEvent>
+#include <QObject>
+
+class KeyEventObserver : public QObject {
+	Q_OBJECT
+
+private:
+	QEvent::Type m_eventType;
+	bool m_consume;
+	std::set< Qt::Key > m_keys;
```

Comment:
While a set is what we want semantically, a `std::vector` will (almost certainly) be more performant in practice given the expected size of `m_keys` due to cache locality. Aka: The O(1) lookup time for the set will be beaten by the O(N) lookup time for the vector.

Even better though would be to turn this into a class templated on the amount of keys to observe as we can use a `std::array` in that case, which is likely optimal for our needs (I am expecting that we'll only use this for a couple of keys and not like 20+).

In the end, I'm not sure whether this is a worthwhile optimization though... It's just that I get a bit itchy when it comes to processes that can delay further input processing as we've had issues with that in the past :eyes: 
That is to say that if you think that the current version is better for readability's sake, then feel free to ignore my comment :)

Reply:
> Even better though would be to turn this into a class templated on the amount of keys to observe as we can use a `std::array` in that case, which is likely optimal for our needs (I am expecting that we'll only use this for a couple of keys and not like 20+).

Error: Template classes not supported by Q_OBJECT :upside_down_face:

Reply:
Oh well... ^^